grakn-graph/src/main/java/ai/grakn/graph/internal/ConceptImpl.java

```diff
@@ -146,10 +142,18 @@ public ConceptId getId(){
     }
 
     /**
+     * Helper method to cast interfaces into their implementation.
      *
-     * @return The hash code of the underlying vertex
+     * @param concept The interface to be casted
+     * @param <X> The type of the implementation to be returned
+     * @return The concept casted to it's implementation
      */
-    public int hashCode() {
+    @SuppressWarnings("unchecked")
+    public static <X extends ConceptImpl> X from(Concept concept){
+        return (X) concept;
+    }
```

Comment:
I'm gonna dump my thoughts about this thing here:

Ideally I'd suggest making sure these cast methods are always safe. If you want to perform a potentially-unsafe cast then you should do it with a manual cast.

We can make them safe if we remove the generic and we only put them in places where we know there is just one valid implementation. For example `ResourceImpl.from(Resource)`, `ConceptVertex.from(Concept)` etc.